File: stack/1047_remove_all_adjacent_duplicates_in_string/1047_remove_all_adjacent_duplicates_in_string.go
package main

import "fmt"

func removeDuplicates(S string) string {
	ans := make([]byte, 0, len(S))
	for _, v := range []byte(S) {
		if len(ans) > 0 && v == ans[len(ans)-1] {
			ans = ans[:len(ans)-1]
			continue
		}
		ans = append(ans, v)
	}

	return string(ans)
}

func main() {
	S := "abbaca"
	res := removeDuplicates(S)
	fmt.Println(res)
}

File: stack/1047_remove_all_adjacent_duplicates_in_string/1047_remove_all_adjacent_duplicates_in_string.py
class Solution:
    def removeDuplicates(self, S: str) -> str:
        res = []
        for i in S:
            if len(res) >= 1 and i == res[-1]:
                res.pop()
                continue
            res.append(i)
        return ''.join(res)


S = "abbaca"
res = Solution().removeDuplicates(S)
print(res)